src/explorer/siteNodeBase.ts

```diff
@@ -253,4 +257,26 @@ export class SiteNodeBase extends NodeBase {
             throw new Error(`Azure Remote Repo is current with ${repo}`);
         }
     }
+
+    private async showWorkspaceFoldersQuickPick(): Promise<string> {
+        const folderQuickPickItems = workspace.workspaceFolders.map((value) => {
+            {
+                return <QuickPickItemWithData<WorkspaceFolder>>{
+                    label: value.name,
+                    description: '',
+                    data: value
+                }
+            }
+        });
+
+        const folderQuickPickOption = { placeHolder: `Select the folder to Local Git deploy.` };
+        const pickedItem = folderQuickPickItems.length == 1 ?
+            folderQuickPickItems[0] : await window.showQuickPick(folderQuickPickItems, folderQuickPickOption);
+
+        if (!pickedItem) {
+            throw new UserCancelledError();
```

Comment:
When this function becomes shared code, I think it would be better to just return undefined instead of throwing UserCancelledError. Let the caller decide whether or not to throw what error.